lib/internal/filter.js

```diff
@@ -1,31 +1,19 @@
-import arrayMap from 'lodash/_arrayMap';
-import property from 'lodash/_baseProperty';
+import filter from 'lodash/_arrayFilter';
 import noop from 'lodash/noop';
 import once from './once';
 
 export default function _filter(eachfn, arr, iteratee, callback) {
     callback = once(callback || noop);
-    var results = [];
+    var truthy = [];
```

Comment:
The creation of the pre-filtered indexes is hard (`[1, 5, 6]`). This is why I went with a true/false at every index, the order is easy because indexes don't need to change.